core/src/main/scala/frameless/CatalystVariance.scala

```diff
@@ -0,0 +1,15 @@
+package frameless
+
+/**
+  * When taking variance, Spark doesn't change this type:
```

Comment:
Might want to add [this comment](https://github.com/adelbertc/frameless/pull/81/files#diff-02d63fc7a767ef03927ba65b737f33feR43) here